rust/private/rust_analyzer.bzl

```diff
@@ -116,51 +118,79 @@ def _crate_id(crate_info):
     """
     return "ID-" + crate_info.root.path
 
-def create_crate(ctx, info, crate_mapping):
-    """Creates a crate in the rust-project.json format
+_DepRefInfo = provider(
+    doc = "RustAnalyzerInfo holds rust crate metadata for targets",
+    fields = {
+        "crate": "BuildInfo: build info for this crate if present",
+        "name": "List[String]: features or other compilation --cfg settings",
```

Comment:
nit, maybe this field could be balled `cfg_flags`, `flags` or `settings`?

Reply:
I actually realized that even though lists and dicts are not hashable in Starlark, tuples are. Now I'm using an inline dict comprehension instead of collections.dict and I don't need an explicit provider type anymore.